src/containers/App/App.js

```diff
@@ -1,18 +1,33 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { Router } from '@reach/router';
 import { I18nProvider } from '@lingui/react';
 import { catalogs } from 'config/i18n';
 import { Home, Login } from 'pages';
+import Callback from 'components/callback/Callback';
+import auth0Client from 'components/auth/Auth';
 
-const App = () => {
-  return (
-    <I18nProvider language="en" catalogs={catalogs}>
-      <Router>
-        <Home path="/" />
-        <Login path="/auth/login" />
-      </Router>
-    </I18nProvider>
-  );
-};
+class App extends Component {
+  async componentDidMount() {
+    if (this.props.location.pathname === '/callback') return;
+    try {
+      await auth0Client.silentAuth();
+      this.forceUpdate();
```

Comment:
Since this part is React, think declaratively. If we have a state of `{ loggedIn: false }`, we can update that state to `{ loggedIn: true }` here instead of forcing an update, and then we can redirect them to either `Home` or `Login` based on that. We may be able to do without the `Callback` component entirely.

``` js
async componentDidMount() {
  if (!this.state.loggedIn) {
    try {
      await auth0Client.silentAuth();
      this.setState({ loggedIn: true });
    } catch (err) {
      // do error things here
    }
  }
}
```

Of course, this doesn't account for session validation or anything like that. `loggedIn` doesn't necessarily mean they're authed.